src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -1,186 +1,226 @@
-                }
-                Logger.Close();
-                throw;
-            }
-
-            Logger.Log("Done");
-            Logger.Close();
-        }
-
-        public void Initialize(GeneratorInitializationContext context)
-        {
-        }
-    }
-}
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using System.Text;
+
+namespace Generator
+{
+    [Generator]
+    public class SourceGenerator : ISourceGenerator
+    {
+        private string _tempFolder;
+
+        private static string GetAssemblyName(GeneratorExecutionContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyName", out var assemblyName);
+            return assemblyName;
+        }
+
+        private static string GetAssemblyVersion(GeneratorExecutionContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.AssemblyVersion", out var assemblyVersion);
+            return assemblyVersion;
+        }
+
+        internal static string GetGeneratedFilesDir(GeneratorExecutionContext context)
+        {
+            // TODO: determine correct location to write to.
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.GeneratedFilesDir", out var generatedFilesDir);
+            Directory.CreateDirectory(generatedFilesDir);
+            return generatedFilesDir;
+        }
+
+        private static bool IsCsWinRTComponent(GeneratorExecutionContext context)
+        {
+            if (context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTComponent", out var isCsWinRTComponentStr))
+            {
+                return bool.TryParse(isCsWinRTComponentStr, out var isCsWinRTComponent) && isCsWinRTComponent;
+            }
+
+            return false;
+        }
+
+        private static string GetCsWinRTExe(GeneratorExecutionContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTExe", out var cswinrtExe);
+            return cswinrtExe;
+        }
+
+        private static bool GetKeepGeneratedSources(GeneratorExecutionContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTKeepGeneratedSources", out var keepGeneratedSourcesStr);
+            return keepGeneratedSourcesStr != null && bool.TryParse(keepGeneratedSourcesStr, out var keepGeneratedSources) && keepGeneratedSources;
+        }
+
+        private static string GetCsWinRTWindowsMetadata(GeneratorExecutionContext context)
+        {
+            context.AnalyzerConfigOptions.GlobalOptions.TryGetValue("build_property.CsWinRTWindowsMetadata", out var cswinrtWindowsMetadata);

```

Comment:
Do we want a function on contexts and strings that returns the `out var` ? we do this same thing here and in a couple other places (iscswinrtcomponent, getgeneratedfilesdir, ...)

Reply:
I was thinking about that before, but I hadn't as these were just 2 line functions that doing this change will only reduce it by a line and I couldn't use it across the different return types consistently.  If I add more such string reutrn functions, it might be worth considering to do to avoid repeating the TryGetValue call in code.